From 35a4e72b0eef67d4cd70c566ba6bb71837ed8e78 Mon Sep 17 00:00:00 2001
From: "Matwey V. Kornilov" <matwey@sai.msu.ru>
Date: Mon, 9 Feb 2015 21:48:01 +0300
Subject: [PATCH 1/2] pci: spear: Use platform_driver_probe instead of
 platform_driver_register

Use platform_driver_probe instead of platform_driver_register
because the former allows to use probe function placed into __init section
and the driver itself is not support hotplugging (yet?).

Patch-mainline: not yet
Signed-off-by: Matwey V. Kornilov <matwey@sai.msu.ru>
---
 drivers/pci/host/pcie-spear13xx.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/pci/host/pcie-spear13xx.c b/drivers/pci/host/pcie-spear13xx.c
index 866465f..51e1344 100644
--- a/drivers/pci/host/pcie-spear13xx.c
+++ b/drivers/pci/host/pcie-spear13xx.c
@@ -371,7 +371,6 @@ static const struct of_device_id spear13xx_pcie_of_match[] = {
 MODULE_DEVICE_TABLE(of, spear13xx_pcie_of_match);
 
 static struct platform_driver spear13xx_pcie_driver __initdata = {
-	.probe		= spear13xx_pcie_probe,
 	.driver = {
 		.name	= "spear-pcie",
 		.of_match_table = of_match_ptr(spear13xx_pcie_of_match),
@@ -382,7 +381,7 @@ static struct platform_driver spear13xx_pcie_driver __initdata = {
 
 static int __init spear13xx_pcie_init(void)
 {
-	return platform_driver_register(&spear13xx_pcie_driver);
+	return platform_driver_probe(&spear13xx_pcie_driver, spear13xx_pcie_probe);
 }
 module_init(spear13xx_pcie_init);
 
-- 
2.1.4

